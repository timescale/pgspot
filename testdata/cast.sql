-- safe: int is automatically interpreted as pg_catalog.int
SELECT 1::int;

-- safe: int4 is the built-in pg_catalog.int4
SELECT 1::pg_catalog.int4;

-- unsafe: the cast is to a custom type, and the type is unqualified
SELECT 1::custom_type;

-- safe: the cast is to a schema-qualified custom type
SELECT 1::custom_schema.type;

-- unsafe: the destination type is not fully-qualified
SELECT '1'::int4;

-- unsafe: the destination type is not fully-qualified
SELECT int4 '1';

-- unsafe: the destination type is not fully-qualified
SELECT CAST('1' AS int4);
