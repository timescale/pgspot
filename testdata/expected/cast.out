PS017: Unqualified object reference: custom_type in CAST(1 AS custom_type)
PS017: Unqualified object reference: int4 in CAST('1' AS int4)
PS017: Unqualified object reference: int4 in CAST('1' AS int4)
PS017: Unqualified object reference: int4 in CAST('1' AS int4)

Errors: 4 Warnings: 0 Unknown: 0
